src/components/MunicipalCharts/ColumnChart.js

```diff
@@ -290,12 +290,21 @@ export default class ColumnChart {
     }
 
     _colMouseOut(d) {
-
-      this._resetHighlight();
-
-
+        if (!this.colClicked) {
+            this._resetHighlight();
+        }
+        this.colClicked = false;
     }
 
+    _colClick(d) {
+        let self = this;
+        self.colClicked = true;
+        let colId = d.target.__data__.municipality['code']
+        let chartSection = this.chart.config.bindto.split(" ")[0];
```

Comment:
nah this isn't a good idea. We can't make assumptions about the css selector passed as the container. It might just be a single ID like in this storybook https://github.com/OpenUpSA/municipal-money-charts/blob/main/src/stories/ColumnChart.story.js#L329

so what you're trying to do here is share enough information with the event recipient to know which chart section's buttons you should update.

What you could do is to create an event listener for each chart container, at which point you know which buttons you want to toggle, becuase you're creating those buttons in the same place, right?

Then you call dispatchevent on the bar that was clicked, rather than on `document`. The event bubbles from where it was dispatched, so it will only be seen by the listener for that section. So you have no need to pass chartSection, only the information about which column was clicked (the data series)

Reply:
Removed chartSection and no longer passing any additional variables through the event as we can get colId (muni ID) can from the event target element